src/mmvc/api/ITriggerMap.hx

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright (c) 2015 Massive Interactive
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of 
+this software and associated documentation files (the "Software"), to deal in 
+the Software without restriction, including without limitation the rights to 
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies 
+of the Software, and to permit persons to whom the Software is furnished to do 
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all 
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE 
+SOFTWARE.
+*/
+
+package mmvc.api;
+
+interface ITriggerMap
+{
+	function map(trigger:Class<Dynamic>, command:Class<ICommand>):Void;
+	function unmap(trigger:Class<Dynamic>):Void;
+	function dispatch(trigger:{}):Void;
+
```

Comment:
I wonder if it's simpler just to have `map(trigger:Dynamic, command:Class<ICommand>)` and handle the different types internally? The compile time type checking isn't going to tell you very much, other than you're using the wrong method for a particular type...